waltz-tools/src/main/java/com/wepay/waltz/tools/client/ClientCli.java

```diff
@@ -172,20 +166,22 @@ protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
                 WaltzClientConfig waltzClientConfig = getWaltzClientConfig(configFilePath);
 
                 // check optional argument
-                int highWaterMark = cmd.hasOption("high-watermark") ? Integer.parseInt(cmd.getOptionValue("high-watermark")) : DEFAULT_HIGH_WATERMARK;
-                if (highWaterMark < -1) {
-                    throw new IllegalArgumentException("high-watermark must be greater or equals to -1");
-                }
                 int numActivePartitions = cmd.hasOption("num-active-partitions") ? Integer.parseInt(cmd.getOptionValue("num-active-partitions")) : DEFAULT_NUMBER_ACTIVE_PARTITIONS;
                 if (numActivePartitions < 1) {
                     throw new IllegalArgumentException("num-active-partitions must be greater or equals to 1");
                 }
 
-                int numTxnToSubmit = txnPerClient * numClients;
-                int numTxnAppended = highWaterMark + 1;
+                // get number of existing transactions across all partitions
+                long numTxnToSubmit = txnPerClient * numClients;
+                long numExistingTransactions = 0;
+                for (int partitionId = 0; partitionId < numActivePartitions; partitionId++) {
+                    long partitionHighWaterMark = getHighWaterMark(partitionId, waltzClientConfig);
+                    numExistingTransactions += partitionHighWaterMark > -1L ? partitionHighWaterMark + 1 : 0;
+                }
+
                 // each client will receive callback of all transactions
-                int expectNumProducerCallbacks = (numTxnAppended + numTxnToSubmit) * numClients;
-                int expectNumConsumerCallbacks = (numTxnAppended + numTxnToSubmit) * 1;
+                int expectNumProducerCallbacks = toIntExact((numExistingTransactions + numTxnToSubmit) * numClients);
```

Comment:
Since `CountDownLatch` cannot take `long` type as default count, casting to `int` with `toIntExact`, which will throw `ArithmeticException` in case of overflow.